_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -683,5 +683,50 @@ mfxStatus mfxSchedulerCore::DoWork()
 } // mfxStatus mfxSchedulerCore::DoWork()
 
 
+mfxStatus mfxSchedulerCore::FindActiveSyncPoint(const void *pOwner, mfxSyncPoint *pSyncPoint)
+{
+    if (nullptr == pOwner || nullptr == pSyncPoint)
+    {
+        return MFX_ERR_NULL_PTR;
+    }
 
+    mfxTaskHandle handle;
+    handle.handle = 0;
+
+    mfxU32 priority;
+    // enter protected section
+    {
+        std::lock_guard<std::mutex> guard(m_guard);
+
+        for (priority = 0; priority < MFX_PRIORITY_NUMBER; priority += 1)
+        {
+            int type;
+
+            for (type = MFX_TYPE_HARDWARE; type <= MFX_TYPE_SOFTWARE; type += 1)
+            {
+                MFX_SCHEDULER_TASK *pTask = m_pTasks[priority][type];
+
+                // run over the tasks with particular priority
+                while (pTask)
```

Comment:
as for me, ```for```-loop looks more clear:
```
for (MFX_SCHEDULER_TASK *pTask = m_pTasks[priority][type]; pTask; pTask = pTask->pNext)
```

Reply:
what about "type <= MFX_TYPE_SOFTWARE" check for this loop?